Add support for /artist/{id}/albums subresource

Ihsahn opened the pull request.

Ihsahn committed:
Add support for /artist/{id}/albums subresource

maxjoehnk commented on src/client.rs:
```diff
@@ -35,6 +42,14 @@ impl DeezerClient {
+    /// Returns the [`Album`] for Artist with the given id.
+    ///
+    /// [Deezer Api Documentation](https://developers.deezer.com/api/artist/albums)
+    pub async fn artist_albums(&self, id: u64, limit: Option<u32>,
```
For now this should be fine, but I actually want to implement an abstraction for paging similar to the api in https://github.com/maxjoehnk/soundcloud-rs.

maxjoehnk commented on src/client.rs:
```diff
@@ -157,10 +172,32 @@ impl DeezerClient {
+        Ok(res.data)
+    }
+
+    async fn get_with_params<T: DeserializeOwned>(&self, url: &str, query_params: &HashMap<String, String>) -> Result<T> {
```
I would prefer if you could change this to an `Option<&HashMap<String, String>>` (maybe in another method so `get_with_params` keeps its signature) so we don't need `lazy_static` as an extra dependency.

Ihsahn replied:
done :)

Ihsahn committed:
Post review changes:
- remove lazy static dependency
- add get_with_optional_params method with Option<&HashMap<String, String>> parameter

maxjoehnk commented on src/client.rs:
```diff
@@ -157,16 +168,51 @@ impl DeezerClient {
+        let url = format!("{}/{}", BASE_URL, url);
+
+        let mut params: HashMap<String, String> = HashMap::new();
+        if let Some(_limit) = limit {
```
You can actually just use `limit` and `offset` as variable names. The underscore indicated an unused variable so it shouldn't be used here.

Ihsahn replied:
changed

maxjoehnk commented on src/client.rs:
```diff
@@ -157,16 +168,51 @@ impl DeezerClient {
-            .json()
-            .await?;
+            .get(url);
+        if query_params.is_some() {
+            request_builder = request_builder.query(query_params.unwrap());
+        }
```
Here you can use pattern matching instead of unwraping like so:
```suggestion
        if let Some(params) = query_params {
            request_builder = request_builder.query(params);
        }
```

Ihsahn replied:
changed

maxjoehnk reviewed (changes requested):
Two minor changes left, then this should be ready for merge.

Ihsahn committed:
fixed remaining post-review issues
